source/application/GUI/Core/DlgBase.h

```diff
@@ -0,0 +1,138 @@
+/*
+ * (c) 2015- Hourglass Resurrection Team
+ * Hourglass Resurrection is licensed under GPL v2.
+ * Refer to the file COPYING.txt in the project root.
+ */
+
+#pragma once
+
+#include <Windows.h>
+#include <CommCtrl.h>
+
+#include <functional>
+#include <map>
+#include <string>
+#include <vector>
+
+#include "MenuBase.h"
+
+/*
+ * ------------------------------------------------------------------------------------------------
+ *                                   Implementation notes
+ * ------------------------------------------------------------------------------------------------
+ *
+ * Usage is simple. Just inherit DlgBase and remember that everything is in DialogPoints instead of
+ * pixels.
+ *
+ * This construction allows for enum IDs for the object IDs, and it is encouraged to use this
+ * method to declare them in order to avoid ID collisions. It is also encouraged to keep the IDs
+ * private to the class inheriting DlgBase.
+ *
+ * When creating an Indirect dialog, SpawnDialogBox() will return -1 on creation failure and 0 on
+ * success. When creating a Prompt dialog, SpawnDialogBox() will return the result of the dialog
+ * box when it exits.
+ *
+ * The purpose of the overloaded WinAPI calls are to send messages to the window and it's controls,
+ * without needing to know the HWND.
+ * If you provide an external interface to your class, don't just proxy all paramterized calls,
+ * wrap them where it makes sense.
+ * You should have class members that look like this (example code, do not actually use):
+ * bool SendCloseMessage()
+ * {
+ *    SendMessage(WM_CLOSE, 0, 0);
+ *    return true;
+ * }
+ * These overloaded SendMessage and SendDlgItemMessage functions can now fail with LRESULT -1 and
+ * ERROR_CAN_NOT_COMPLETE, this is not a standard error for these functions.
+ *
+ * Note:
+ * Some items like the DropDownList and ListView need further initialization during WM_INITDIALOG
+ * as some of their settings are not exposed as style flags, but need setting using
+ * SendDlgItemMessage().
+ * Use a WM_INITDIALOG callback and perform the init steps in that.
+ */
+
+/*
+ * Undefine some Windows defines to make sure calling these functions sends the call to the DlgBase
+ * implementation.
+ */
+#undef SendMessage
+#undef SendDlgItemMessage
+
+class CallbackBase;
+
+class DlgBase
+{
+protected:
+    enum class DlgMode
+    {
+        PROMPT,
+        INDIRECT,
+    };
+
+    DlgBase(const std::wstring& caption, SHORT x, SHORT y, SHORT w, SHORT h);
+    ~DlgBase();
+
+    //void AddUpDownControl(DWORD id, SHORT x, SHORT y, SHORT w, SHORT h);
+    //void AddDropDownList(DWORD id, SHORT x, SHORT y, SHORT w, SHORT drop_distance);
+    //void AddListView(DWORD id,
+    //                 SHORT x, SHORT y,
+    //                 SHORT w, SHORT h,
+    //                 bool single_selection, bool owner_data);
+    //void AddIPEditControl(DWORD id, SHORT x, SHORT y);

```

Comment:
Maybe delete this stuff out of this PR? And just add it back whenever you add it back later.